crates/re_data_source/src/data_loader/loader_rrd.rs

```diff
@@ -0,0 +1,99 @@
+use re_log_encoding::decoder::Decoder;
+
+// ---
+
+/// Loads data from any `rrd` file or in-memory contents.
+pub struct RrdLoader;
+
+impl crate::DataLoader for RrdLoader {
+    #[inline]
+    fn name(&self) -> String {
+        "rerun.data_loaders.Rrd".into()
+    }
+
+    #[cfg(not(target_arch = "wasm32"))]
+    fn load_from_file(
+        &self,
+        // NOTE: The Store ID comes from the rrd file itself.
+        _store_id: re_log_types::StoreId,
+        filepath: std::path::PathBuf,
+        tx: std::sync::mpsc::Sender<crate::LoadedData>,
+    ) -> Result<(), crate::DataLoaderError> {
+        use anyhow::Context as _;
+
+        re_tracing::profile_function!(filepath.display().to_string());
+
+        let extension = filepath
+            .extension()
+            .unwrap_or_default()
+            .to_ascii_lowercase()
+            .to_string_lossy()
+            .to_string();
+
+        if extension != "rrd" {
+            return Ok(()); // simply not interested
```

Comment:
it seems wrong to me that we use `Ok(())` to mean two very different things:

* I successfully loaded the file
* Unsupported file format / I didn't try to load the file

In the egui data loaders we have a specific error message for unsupported file extensions/types. That's in fact how we find the correct loader: try them all until one doesn't return the `NotSupported ` error

Reply:
For reference: https://docs.rs/egui/latest/egui/load/index.html

Reply:
Note: we don't have a notion of _the one correct loader_ in our case: there can be more than one.

I'll make a follow up PR with `unsupported` errors (for builtin) and exit codes (for externals), with some extra logic to return an actual error in the end if no loader whatsoever could be found.